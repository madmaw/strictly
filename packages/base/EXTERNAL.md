An external file

[home](README.md)