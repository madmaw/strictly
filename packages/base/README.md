# Base

Utility functions and types to support @strictly

[external](EXTERNAL.md)