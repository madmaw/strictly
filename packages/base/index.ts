export * from './errors/not_implemented'
export * from './errors/unexpected_implementation'
export * from './errors/unreachable'
export * from './test'
export * from './types/element_of_array'
export * from './types/exhaustive_array_of_union'
export * from './types/exhaustive_array_of_union'
export * from './types/extract_generics'
export * from './types/is_field_readonly'
export * from './types/maybe'
export * from './types/printable_of'
export * from './types/required_of_record'
export * from './types/reverse'
export * from './types/string_key_of'
export * from './util/array'
export * from './util/cache'
export * from './util/delay'
export * from './util/format'
export * from './util/is_defined'
export * from './util/json'
export * from './util/poll'
export * from './util/preconditions'
export * from './util/promises'
export * from './util/record'
