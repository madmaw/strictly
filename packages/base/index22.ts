export * from './errors/NotImplemented'
export * from './errors/UnexpectedImplementation'
export * from './errors/Unreachable'
export * from './test'
export * from './types/ElementOfArray'
export * from './types/ExhaustiveArrayOfUnion'
export * from './types/ExhaustiveArrayOfUnion'
export * from './types/ExtractGenerics'
export * from './types/IsFieldReadonly'
export * from './types/Maybe'
export * from './types/PrintableOf'
export * from './types/RequiredOfRecord'
export * from './types/Reverse'
export * from './types/StringConcatOf'
export * from './types/StringKeyOf'
export * from './util/array'
export * from './util/Cache'
export * from './util/delay'
export * from './util/format'
export * from './util/isDefined'
export * from './util/json'
export * from './util/poll'
export * from './util/preconditions'
export * from './util/promises'
export * from './util/record'
