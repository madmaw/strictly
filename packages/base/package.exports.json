{
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",

        "default": "./dist/index.js"
      },
      "require": {
        "types": "./dist/index.d.cts",

        "default": "./dist/index.cjs"
      }
    }
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts"
}
