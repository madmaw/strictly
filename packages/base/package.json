{
  "author": "Chris <chris.glover@gmail.com> (@madmaw)",
  "devDependencies": {
    "@strictly/support-vite": "*",
    "json": "^11.0.0"
  },
  "homepage": "https://madmaw.github.io/strictly/base",
  "keywords": [
    "react",
    "state management",
    "types"
  ],
  "license": "MIT",
  "name": "@strictly/base",
  "packageManager": "yarn@1.22.22",
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "directory": "packages/base",
    "type": "git",
    "url": "git+https://github.com/madmaw/strictly.git"
  },
  "scripts": {
    "build": "tsup",
    "check-types": "tsc",
    "clean": "del-cli dist .out .turbo",
    "lint": "cross-env ESLINT_USE_FLAT_CONFIG=false eslint . --max-warnings=0",
    "lint:fix": "cross-env ESLINT_USE_FLAT_CONFIG=false eslint . --fix",
    "release:exports": "json -f package.json -f package.exports.json --merge > package.release.json",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "type": "module",
  "version": "0.0.14"
}
