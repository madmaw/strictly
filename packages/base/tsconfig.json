{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./.out"
  },
  "extends": "../../tsconfig.json",
  "include": [
    "**/*.ts",
    "tsconfig.json"
  ],
  "references": [
    {
      "path": "../../support/vite"
    }
  ]
}
