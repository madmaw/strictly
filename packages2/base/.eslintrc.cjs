/* eslint-env node */
/** @type {import('@types/eslint').Linter.Config} */
module.exports = {
  root: true,
  extends: ['../../.eslintrc.cjs'],
  parserOptions: {
    project: [
      'tsconfig.json',
      'tsconfig.build.json',
    ],
    tsconfigRootDir: __dirname,
    sourceType: 'module',
    ecmaVersion: '2015',
  },
  rules: {
    'import/no-extraneous-dependencies': [
      'warn',
      {
        packageDir: [
          '.',
          '../..',
        ],
      },
    ],
  },
}
