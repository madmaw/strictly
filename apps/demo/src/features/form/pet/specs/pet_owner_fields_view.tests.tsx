import { composeStories } from '@storybook/react'
import { toArray } from '@strictly/base'
import {
  render,
} from '@testing-library/react'
import * as stories from './pet_owner_fields_view.stories'

const composedStories = composeStories(stories)

describe('PetOwnerFieldsView', function () {
  it.each(toArray(composedStories))('renders %s', function (_name, Story) {
    const wrapper = render(<Story />)
    expect(wrapper.container).toMatchSnapshot()
  })
})
