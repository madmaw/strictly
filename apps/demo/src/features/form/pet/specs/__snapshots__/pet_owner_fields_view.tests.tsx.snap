// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PetOwnerFieldsView > renders Empty 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4081bf90 mantine-Group-root"
      data-grow="true"
      style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: start; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0cyk5rcyk"
          id="mantine-0cyk5rcyk-label"
        >
          First Name
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-variant="default"
            id="mantine-0cyk5rcyk"
            name="$.firstName"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0px4bipx4"
          id="mantine-0px4bipx4-label"
        >
          Surname
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-variant="default"
            id="mantine-0px4bipx4"
            name="$.surname"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        data-required="true"
        for="mantine-12voha2vo"
        id="mantine-12voha2vo-label"
      >
        Phone number
        <span
          aria-hidden="true"
          class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-variant="default"
          id="mantine-12voha2vo"
          name="$.phoneNumber"
          placeholder="04xxxxxxxx"
          required=""
          type="tel"
          value=""
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-TextInput-section"
          data-position="right"
        >
          <button
            class="mantine-focus-auto mantine-active m_220c80f2 m_86a44da5 mantine-CloseButton-root m_87cf2631 mantine-UnstyledButton-root"
            data-variant="subtle"
            type="button"
          >
            <svg
              fill="none"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        for="mantine-1fu8n1fu8"
        id="mantine-1fu8n1fu8-label"
      >
        Email
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-variant="default"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-variant="default"
          id="mantine-1fu8n1fu8"
          name="$.email"
          placeholder="me@somewhere.org"
          type="email"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PetOwnerFieldsView > renders Errors 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4081bf90 mantine-Group-root"
      data-grow="true"
      style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: start; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        data-error="true"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0cyk5rcyk"
          id="mantine-0cyk5rcyk-label"
        >
          First Name
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-error="true"
          data-variant="default"
          style="--input-margin-bottom: calc(var(--mantine-spacing-xs) / 2);"
        >
          <input
            aria-describedby="mantine-0cyk5rcyk-error"
            aria-invalid="true"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-error="true"
            data-variant="default"
            id="mantine-0cyk5rcyk"
            name="$.firstName"
            type="text"
            value="Bob"
          />
        </div>
        <p
          class="m_8f816625 mantine-InputWrapper-error mantine-TextInput-error"
          id="mantine-0cyk5rcyk-error"
        >
          First name must be at least 3 characters long
        </p>
      </div>
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        data-error="true"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0px4bipx4"
          id="mantine-0px4bipx4-label"
        >
          Surname
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-error="true"
          data-variant="default"
          style="--input-margin-bottom: calc(var(--mantine-spacing-xs) / 2);"
        >
          <input
            aria-describedby="mantine-0px4bipx4-error"
            aria-invalid="true"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-error="true"
            data-variant="default"
            id="mantine-0px4bipx4"
            name="$.surname"
            type="text"
            value="Jane"
          />
        </div>
        <p
          class="m_8f816625 mantine-InputWrapper-error mantine-TextInput-error"
          id="mantine-0px4bipx4-error"
        >
          Surname must be at least 3 characters long
        </p>
      </div>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      data-error="true"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        data-required="true"
        for="mantine-12voha2vo"
        id="mantine-12voha2vo-label"
      >
        Phone number
        <span
          aria-hidden="true"
          class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-error="true"
        data-variant="default"
        data-with-right-section="true"
        style="--input-margin-bottom: calc(var(--mantine-spacing-xs) / 2);"
      >
        <input
          aria-describedby="mantine-12voha2vo-error"
          aria-invalid="true"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-error="true"
          data-variant="default"
          id="mantine-12voha2vo"
          name="$.phoneNumber"
          placeholder="04xxxxxxxx"
          required=""
          type="tel"
          value="005000500"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-TextInput-section"
          data-position="right"
        >
          <button
            class="mantine-focus-auto mantine-active m_220c80f2 m_86a44da5 mantine-CloseButton-root m_87cf2631 mantine-UnstyledButton-root"
            data-variant="subtle"
            type="button"
          >
            <svg
              fill="none"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="m_8f816625 mantine-InputWrapper-error mantine-TextInput-error"
        id="mantine-12voha2vo-error"
      >
        Must be a valid phone number
      </p>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      data-error="true"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        for="mantine-1fu8n1fu8"
        id="mantine-1fu8n1fu8-label"
      >
        Email
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-error="true"
        data-variant="default"
        style="--input-margin-bottom: calc(var(--mantine-spacing-xs) / 2);"
      >
        <input
          aria-describedby="mantine-1fu8n1fu8-error"
          aria-invalid="true"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-error="true"
          data-variant="default"
          id="mantine-1fu8n1fu8"
          name="$.email"
          placeholder="me@somewhere.org"
          type="email"
          value="email@e.mail"
        />
      </div>
      <p
        class="m_8f816625 mantine-InputWrapper-error mantine-TextInput-error"
        id="mantine-1fu8n1fu8-error"
      >
        Must be a valid email
      </p>
    </div>
  </div>
</div>
`;

exports[`PetOwnerFieldsView > renders Populated 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
  >
    <div
      class="m_4081bf90 mantine-Group-root"
      data-grow="true"
      style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: start; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0px4bipx4"
          id="mantine-0px4bipx4-label"
        >
          First Name
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-variant="default"
            id="mantine-0px4bipx4"
            name="$.firstName"
            type="text"
            value="Bob"
          />
        </div>
      </div>
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-12voha2vo"
          id="mantine-12voha2vo-label"
        >
          Surname
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            autocapitalize="words"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-variant="default"
            id="mantine-12voha2vo"
            name="$.surname"
            type="text"
            value="Jane"
          />
        </div>
      </div>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        data-required="true"
        for="mantine-1fu8n1fu8"
        id="mantine-1fu8n1fu8-label"
      >
        Phone number
        <span
          aria-hidden="true"
          class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-variant="default"
          id="mantine-1fu8n1fu8"
          name="$.phoneNumber"
          placeholder="04xxxxxxxx"
          required=""
          type="tel"
          value="005000500"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-TextInput-section"
          data-position="right"
        >
          <button
            class="mantine-focus-auto mantine-active m_220c80f2 m_86a44da5 mantine-CloseButton-root m_87cf2631 mantine-UnstyledButton-root"
            data-variant="subtle"
            type="button"
          >
            <svg
              fill="none"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
        for="mantine-1ssssssss"
        id="mantine-1ssssssss-label"
      >
        Email
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
        data-variant="default"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
          data-variant="default"
          id="mantine-1ssssssss"
          name="$.email"
          placeholder="me@somewhere.org"
          type="email"
          value="email@e.mail"
        />
      </div>
    </div>
  </div>
</div>
`;
